src/pages/Index.js

```diff
@@ -65,54 +120,164 @@ modalWithImage.setEventListeners();
 /*                                       */
 /*             UserInfo                  */
 /*                                       */
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => {
+    console.error("Failed to load user information:", err);
+  });
 
 const userInfo = new UserInfo({
   profileTitle: ".profile__title",
   profileDescription: ".profile__description",
+  avatarSelector: ".profile__avatar",
 });
 
 /*                                       */
-/*             card.js                   */
+/*           ProfileEditModal            */
 /*                                       */
 
-function renderCard(cardData) {
-  const addCard = new Card(cardData, cardSelector, handleImageClick);
-  return addCard.getView();
+const editModal = new ModalWithForm(
+  "#profile-edit-modal",
+  handleProfileEditSubmit
+);
+
+editModal.setEventListeners();
+
+function handleProfileEditSubmit(formData) {
+  editModal.renderLoading(true);
+  api
+    .updateProfileInfo({ name: formData.title, about: formData.description })
+    .then(() => {
+      userInfo.setUserInfo({
+        name: formData.title,
+        description: formData.description,
+        avatar: userInfo.getUserInfo().avatar,
+      });
+      editModal.close();
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      editModal.renderLoading(false);
+    });
 }
 
-const cardSection = new Section(
-  {
-    items: initialCards,
-    settings,
-    renderer: renderCard,
-  },
+/*                                       */
+/*              avatar                   */
+/*                                       */
+
+function handleAvatarSubmit(data) {
+  editAvatarModal.renderLoading(true);
+  api
+    .updateAvatar({ avatar: data.link })
+    .then((res) => {
+      userInfo.updatedAvatar(res.avatar);
+      // formValidators["edit-avatar-form"].disableButton();
+      // formValidators["edit-avatar-form"].resetForms();
+      editAvatarModal.close();
+    })
+    .catch((err) => {
+      console.error("Failed to update user avatar:", err);
+    })
+    .finally(() => {
+      editAvatarModal.renderLoading(false);
+    });
+}
 
-  ".cards__list"
+const editAvatarModal = new ModalWithForm(
+  "#edit-avatar-modal",
+  handleAvatarSubmit
 );
+editAvatarModal.setEventListeners();
 
-cardSection.renderItems();
+profileAvatarContainer.addEventListener("click", () => {
+  editAvatarModal.open();
+});
 
 /*                                       */
-/*             Event Handlers            */
+/*          LIKE & DISLIKE               */
 /*                                       */
 
+function handleLikeClick(card) {
+  if (card.setIsLiked === true) {
```

Comment:
Needs correction: please don't check that the setter exists, here you should check the likes state. create a method `getIsLiked` in the crd class and use it here:

```js
// Card.js

getIsLiked() {
  return this._isLiked;
}

// here
if (card.getIsLiked()) { .. }
```